app/controllers/actions/DataRetrievalAction.scala

```diff
@@ -19,17 +19,18 @@ package controllers.actions
 import javax.inject.Inject
 import models.requests.{IdentifierRequest, OptionalDataRequest}
 import play.api.mvc.ActionTransformer
-import repositories.SessionRepository
+import repositories.AbstractRepository
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.HeaderCarrierConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class DataRetrievalActionImpl @Inject()(val sessionRepository: SessionRepository)(
+class DataRetrievalActionImpl @Inject()(val sessionRepository: AbstractRepository)(
```

Comment:
this should be sessionRepository: SessionRepository

Reply:
this is injected as session using injections
bind(classOf[AbstractRepository]).to(classOf[SessionRepository]).asEagerSingleton()